src/GoogleCloudStorageServiceProvider.php

```diff
@@ -85,19 +85,15 @@ public function boot()
     private function createClient($config)
     {
         $keyFile = array_get($config, 'key_file');
-        if (is_string($keyFile)) {
+        if (is_array($keyFile)) {
             return new StorageClient([
                 'projectId' => $config['project_id'],
-                'keyFilePath' => $keyFile,
+                'keyFile' => array_merge(["project_id" => $config['project_id']], $keyFile)
```

Comment:
we can actually skip this whole `is_array` check entirely and just remove the `array_merge`. The key file should already contain `project_id`, so I'm not sure why it needs to be merged in the first place?